### PR TITLE
feat(investments): contribution-adjusted returns

### LINKED ISSUE #401: feat(investments): contribution-adjusted returns
## Motivation

Net-worth charts mix deposits, withdrawals, and market movement. Copilot-style investment performance excludes transfers so users see whether investments performed well, not only whether they added money.

## Acceptance criteria

- Compute contribution-adjusted return per account, wrapper, category, and household
- Distinguish deposits, withdrawals, fees, dividends, and valuation change
- Show balance line and return line separately
- Add period selector: 1M / 3M / YTD / 1Y / All
- Document formula: money-weighted return first, time-weighted optional later
- Tests cover deposits that do not count as gains

### PR DESCRIPTION
**Pull request by Automaat**
## Motivation

Net-worth charts mix deposits, withdrawals and market movement so users see whether they added more money, not whether their investments actually performed.

## Implementation information

- Pure `XIRR` function in `internal/investment/xirr.go` — Newton-Raphson on dated cash flows, returns money-weighted (annualized) return. Handles deposit-only edge case (no negative flow) and convergence failure as `ErrXIRRNoConverge`. 6 unit tests cover buy-and-hold, staircase contributions, deposits-not-counting-as-gains, deposits-alone-no-convergence, and SimpleROI.
- New `GET /api/investment/returns?scope=&id=&value=&period=` endpoint. Scope = `all|account|category|wrapper`. Period = `1m|3m|ytd|1y|all`. Response carries deposits, withdrawals, net contributed, current value, valuation change, simple ROI %, money-weighted %, plus `has_snapshot` and `convergence_failed` flags so the UI can degrade gracefully when XIRR can't compute.
- Store gains `ScopedFlows` and `LatestValueInScope`, parameter-built so the same query handles account/category/wrapper/all.
- New `ContributionAdjustedReturns.svelte` widget with the 1M/3M/YTD/1Y/All period selector. Three widgets mounted on `/metryki` (household, stocks, bonds) so the user sees both naive ROI and XIRR side by side.
- Formula documented at the top of `xirr.go`. Time-weighted return is the documented next step — out of scope for this PR.

Closes #401

> Changelog: feat(investments): contribution-adjusted returns

### CODE REVIEW

File: backend-go/internal/investment/store.go [hunk 1]
```diff
@@ -23,6 +26,128 @@ type CategoryStats struct {
...
+}
+
+// PeriodWindow returns (since, asOf). A nil since means "since the first cash
+// flow in scope" (= "All").
+type PeriodWindow struct {
+	Since *time.Time
+	AsOf  time.Time
+}
+
+// ScopedFlows pulls all active transactions in scope between `since` (open)
```
[the remark this answers is not in the document]
Comment (Automaat) [reply backend-go/internal/investment/store.go]: Fixed — comment now says both bounds are inclusive. bbe091b

File: backend-go/internal/investment/handler.go [hunk 2]
```diff
@@ -31,6 +32,161 @@ func NewHandler(store *Store, logger *slog.Logger) *Handler {
...
+	}
+	netContrib := deposits - withdrawals
+	currentValue, _ := value.Float64()
+	valuationChange := currentValue - netContrib
+
+	resp := returnsResponse{
+		Scope:           scopeWire,
+		Period:          period,
+		AsOf:            now.Format("2006-01-02"),
+		Deposits:        pyFloat(deposits),
```
[the remark this answers is not in the document]
Comment (Automaat) [reply backend-go/internal/investment/handler.go:149]: as_of now reports snapDate when a snapshot exists; falls back to now only when there are no snapshots. bbe091b

File: backend-go/internal/investment/handler.go [hunk 3]
```diff
@@ -31,6 +32,161 @@ func NewHandler(store *Store, logger *slog.Logger) *Handler {
...
+		writeDetailError(w, http.StatusBadRequest, vErr)
+		return
+	}
+	period := r.URL.Query().Get("period")
+	if period == "" {
+		period = "all"
+	}
+	now := time.Now().UTC()
+	window := windowFor(period, now)
+
```
[the remark this answers is not in the document]
Comment (Automaat) [reply backend-go/internal/investment/handler.go:85]: For windowed periods, the handler now fetches the snapshot at `since` and injects it as an opening positive cash flow into the XIRR series. valuation_change and simple_roi_pct are adjusted to count opening + net flows as the basis. bbe091b

File: backend-go/internal/investment/handler.go [hunk 4]
```diff
@@ -31,6 +32,161 @@ func NewHandler(store *Store, logger *slog.Logger) *Handler {
...
+// or whether they only added more money.
+func (h *Handler) Returns(w http.ResponseWriter, r *http.Request) {
+	scope, scopeWire, vErr := parseScope(r)
+	if vErr != "" {
+		writeDetailError(w, http.StatusBadRequest, vErr)
+		return
+	}
+	period := r.URL.Query().Get("period")
+	if period == "" {
+		period = "all"
```
[the remark this answers is not in the document]
Comment (Automaat) [reply backend-go/internal/investment/handler.go:77]: Period now validates against the same set as the docs and returns 400 on invalid input. bbe091b

File: backend-go/internal/investment/handler.go [hunk 5]
```diff
@@ -31,6 +32,161 @@ func NewHandler(store *Store, logger *slog.Logger) *Handler {
...
+	}
+	if window.Since != nil {
+		s := window.Since.Format("2006-01-02")
+		resp.Since = &s
+	}
+	if hasSnap && currentValue > 0 {
+		// Closing terminal flow = liquidate at current value at the snapshot date.
+		xirrFlows = append(xirrFlows, CashFlow{Date: snapDate, Amount: -currentValue})
+		mwr, xerr := XIRR(xirrFlows)
+		if xerr != nil {
```
[the remark this answers is not in the document]
Comment (Automaat) [reply backend-go/internal/investment/handler.go]: Dropped the `currentValue > 0` guard — XIRR is attempted whenever hasSnap is true; failures fall through to convergence_failed. bbe091b

File: src/lib/types/api.generated.ts [hunk 6]
```diff
@@ -2979,6 +2979,67 @@ export interface paths {
         patch?: never;
         trace?: never;
     };
+    "/api/investment/returns": {
+        parameters: {
+            query?: never;
+            header?: never;
+            path?: never;
```
[the remark this answers is not in the document]
Comment (Automaat) [reply src/lib/types/api.generated.ts:2986]: Acknowledged. internal/apispec/Route lacks a query-params field today, so the generator can't emit them without a framework change. Tracked separately — would prefer to extend apispec across all packages in one PR rather than special-case this endpoint.

File: backend-go/internal/investment/handler.go [hunk 7]
```diff
@@ -31,6 +32,161 @@ func NewHandler(store *Store, logger *slog.Logger) *Handler {
...
+	window := windowFor(period, now)
+
+	value, snapDate, hasSnap, err := h.store.LatestValueInScope(r.Context(), scope, now)
+	if err != nil {
+		h.logger.Error("returns: snapshot lookup", "err", err)
+		writeDetailError(w, http.StatusInternalServerError, "Internal Server Error")
+		return
+	}
+	flows, err := h.store.ScopedFlows(r.Context(), scope, PeriodWindow{Since: window.Since, AsOf: now})
+	if err != nil {
```
[the remark this answers is not in the document]
Comment (Automaat) [reply backend-go/internal/investment/handler.go]: Flows are now upper-bounded at snapDate when a snapshot exists, so post-snapshot transactions don't pollute the return. bbe091b
